rllib/algorithms/algorithm_config.py

```diff
@@ -2930,6 +2930,31 @@ def get_marl_module_spec(
                         "is passed in nor in the default module spec used in "
                         "the algorithm."
                     )
+            if module_spec.catalog_class is None:
+                if isinstance(default_spec, SingleAgentRLModuleSpec):
+                    module_spec.catalog_class = default_spec.catalog_class
+                elif isinstance(default_spec.module_specs, SingleAgentRLModuleSpec):
+                    catalog_class = default_spec.module_specs.catalog_class
+                    # This should be already checked in validate() but we check it
+                    # again here just in case
+                    if catalog_class is None:
+                        raise ValueError(
+                            "The default rl_module spec cannot have an empty "
```

Comment:
the default marl module spec *. Right? I assume we only enter this branch if the `default_spec` is a `MultiAgentRLModuleSpec`

Reply:
Not really. This is the RLModule spec with in the default MARLSpec.